miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/RunPage.java

```diff
@@ -0,0 +1,227 @@
+    checkBoxesAndSelectOption(partitions, "Assign Pool", option);
+    switch (option) {
+    case PoolSearch.NO_POOL:
+      break;
+    case PoolSearch.SEARCH:
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Search for Pool to Assign"));
+      dialog.findElement(By.tagName("input")).sendKeys(searchText);
+      getDriver().findElement(By.id("ok")).click();
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Select Pool"));
+      if (assignFirstPool) clickFirstPoolTile(true, searchText);
+      break;
+    case PoolSearch.OUTSTANDING_MATCH:
+    case PoolSearch.OUTSTANDING_ALL:
+    case PoolSearch.RECENT:
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Select Pool"));
+      if (assignFirstPool) clickFirstPoolTile(false, searchText);
+      break;
+    case PoolSearch.RTR:
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Select Pool"));
+      if (assignFirstPool) clickFirstPoolTile(true, searchText);
+      break;
+    }
+  }
+
+  public RunPage assignPools(List<Integer> partitions, String option, String searchText) {
+    WebElement html = getHtmlElement();
+    searchForPools(true, partitions, option, searchText);
+    waitForPageRefresh(html);
+    return new RunPage(getDriver());
+  }
+
+  public RunPage setPartitionQC(List<Integer> partitions, String option, String noteText) {
+    WebElement html = getHtmlElement();
+    checkBoxesAndSelectOption(partitions, "Set QC", option);
+    switch (option) {
+    case LaneQC.OK:
+    case LaneQC.OK_COLLAB:
+    case LaneQC.FAIL_INSTRUMENT:
+    case LaneQC.FAIL_LIB_PREP:
+    case LaneQC.FAIL_ANALYSIS:
+      break;
+    case LaneQC.FAIL_OTHER:
+    case LaneQC.FAIL_SEQINFO:
+    case LaneQC.FAIL_SEQOPS:
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Failed: Other problem Notes"));
+      dialog.findElement(By.tagName("input")).sendKeys(noteText);
+      getDriver().findElement(By.id("ok")).click();
+      break;
+    }
+    waitForPageRefresh(html);
+    return new RunPage(getDriver());
+  }
+
+  public String getLaneInfo(String columnHeading, int rowNum) {
+    DataTable partitionsTable = new DataTable(partitionsSection.findElement(By.id("list_partition")));
+    return partitionsTable.getTextAtCell(columnHeading, rowNum);
+  }
+
+  private void checkBoxesAndSelectOption(List<Integer> partitions, String titleText, String option) {
+    DataTable partitionsTable = new DataTable(partitionsSection.findElement(By.id("list_partition")));
+    partitions.forEach(partition -> partitionsTable.checkBoxForRow(partition));
+    partitionsSection.findElement(By.linkText(titleText)).click();
+    waitUntil(visibilityOf(dialog));
+    waitUntil(textToBe(By.className("ui-dialog-title"), titleText));
+    dialog.findElement(By.linkText(option)).click();
+  }
+
+  private void clickFirstPoolTile(boolean extraWait, String searchText) {
+    if (extraWait) {
+      waitUntil(textToBe(By.className("ui-dialog-title"), "Select Pool"), 60);
```

Comment:
Wow, searching by "ready to run" is pretty slow for some reason...

Reply:
Yeah, I'm not sure why that is.